cli/azd/test/functional/package_test.go

```diff
@@ -69,6 +69,48 @@ func Test_CLI_Package_FromServiceDirectory(t *testing.T) {
 	require.Contains(t, result.Stdout, "Packaging service web")
 }
 
+func Test_CLI_Package_FromServiceDirectorySymlinked(t *testing.T) {
```

Comment:
This is really useful! I'm going to have to take a second look here on the proposed fix. From what I gander, the proposed fix *does* solve the particular case of python packaging... but there are still edge cases with how `azd` does zip deploy for other languages where a `copy.Copy` isn't performed.

I'm seeing if I can use this test to verify a fix in `rzip.CreateFromDirectory` to do a deep copy.. Supposedly for the python code path, we'd end up doing a shallow copy in our current behavior, but a deep copy when it's time to zip for deployment.

Reply:
I did end up submitting #4773 which I confirmed locally passes this test. What happens underneath the covers is that `azd` will do a shallow copy with the symlinks, but during zip-time, it would resolve the symbolically linked directories -- it also happens to be the source of where the reported error come from.

@jjgriff93 feel free to see if the private bits there (see install instructions [here](https://github.com/Azure/azure-dev/pull/4773#issuecomment-2644072908)) also happen to addresss your scenario. I do appreciate the effort spent here to document and propose the changes -- hope it helps!

Reply:
Just tested and working, thanks @weikanglim! Shall close as fixed now that https://github.com/Azure/azure-dev/pull/4773 is merged